lib/CompilerImpl/ClangLibCompiler.cpp

```diff
@@ -316,8 +346,10 @@ ClangLibCompiler::runOptimizer(const std::filesystem::path &src_IR,
   std::unique_ptr<llvm::TargetMachine> actualTM(optTMachine);
   // Override function attributes based on CPUStr, FeaturesStr,
   // and command line flags.
+  // This also causes the program to segfault if executed, hence it is commented out
+#if LLVM_VERSION_MAJOR < 18
   llvm::codegen::setFunctionAttributes(optCPUStr, optFeaturesStr, *module);
-
+#endif
```

Comment:
on which version?

Reply:
llvm 18 and 19

Reply:
If it works on 17, we can keep it, otherwise remove it entirely